app/renderer/bookmarks.js

```diff
@@ -0,0 +1,512 @@
+/*
+* Vieb - Vim Inspired Electron Browser
+* Copyright (C) 2019-2025 Jelmer van Arnhem
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+"use strict"
+
+const {ipcRenderer} = require("electron")
+const {
+    appData,
+    getSetting,
+    isAbsolutePath,
+    isFile,
+    joinPath,
+    notify,
+    readJSON,
+    specialChars,
+    specialCharsAllowSpaces,
+    writeJSON
+} = require("../util")
+const {
+    currentPage,
+    currentTab
+} = require("./common")
+
+/**
+ * @typedef {{
+ *  keywords: string[],
+ *  name: string,
+ *  path: string
+ * }} Folder
+ */
+
+/**
+ * @typedef {{
+ *  id: number,
+ *  name: string,
+ *  title: string,
+ *  url: string,
+ *  path: string,
+ *  keywords: string[],
+ *  bg?: string,
+ *  fg?: string,
+ *  [key: string]: any
+ * }} Bookmark
+ */
+
+/**
+ * @typedef {{
+ *   bookmarks: Bookmark[],
+ *   folders: Folder[],
+ *   lastId: number,
+ * }} BookmarkData
+ */
+
+/** @type {BookmarkData} */
+let bookmarkData = {
+    "bookmarks": [],
+    "folders": [],
+    "lastId": 0
+}
+let bookmarksFile = ""
+const validFoldersOptions = [
+    "keywords",
+    "name",
+    "path"
+]
+const validBookmarkOptions = [
+    "bg",
+    "fg",
+    "id",
+    "keywords",
+    "name",
+    "path",
+    "title",
+    "url"
+]
+
+/**
+ * Returns bookmarkData object.
+ * @returns {BookmarkData}
+ */
+const getBookmarkData = () => bookmarkData
+
+/**
+ * Adds folder to the bookmarkData for the given path.
+ * @param {string} path - The path of the folder to add.
+ */
+const addFolder = path => {
+    const currentStoredPaths = bookmarkData.folders.map(f => f.path)
+    if (!currentStoredPaths?.includes(path)) {
+        bookmarkData.folders.push(
+            {"keywords": [], "name": "", path})
+    }
+}
+
+
+/**
+ * Write JSON to file.
+ */
+const writeBookmarksToFile = () => {
+    writeJSON(bookmarksFile, bookmarkData)
+}
+
+/**
+ * Fix bookmarkData values based on the option type.
+ * @param {string} option - The option to fix.
+ * @param {string} value - The value to fix.
+ */
+const fixBookmarkData = (option, value) => {
+    let correctFormat = ""
```

Comment:
Since this value is only set, never modified, and not reused, I would prefer if it wasn't there and the results in each if are returned directly. Or, in case of reworking the function below, it might even be more readable without this function, as the functionality is completely different per option type.